core/kernel/user_ta.c

```diff
@@ -54,7 +54,13 @@ static void init_utee_param(struct utee_params *up,
 {
 	size_t n;
 
-	up->types = p->types;
+	/*
+	 * NOTE: The type of up->types is uint64_t, while the type of p->types
+	 * is uint32_t. Hence, to avoid unexpected behaviors
+	 */
+	clear_user(&up->types, sizeof(up->types));
+	copy_to_user(&up->types, &p->types, sizeof(p->types));
```

Comment:
test resturn value?

Reply:
shall I change the function return type?

Reply:
yes, i think it needs to